src/options.cpp

```diff
@@ -160,6 +169,24 @@ bool options::dump_errors(output_stream& out) const {
     out.append_copy(
         u8"warning: multiple standard input given on command line\n"sv);
   }
+
+  if (this->has_vim_file_bufnr && this->lsp_server) {
+    out.append_copy(
+        u8"warning: ignoring --vim-file-bufnr in --lsp-server mode\n"sv);
+  } else if (this->has_vim_file_bufnr &&
+             this->output_format != output_format::vim_qflist_json) {
+    out.append_copy(
+        u8"warning: --output-format selected which doesn't use --vim-file-bufnr\n"sv);
+  } else {
+    for (const auto& bufnr : this->warning_vim_bufnr_without_file) {
+      out.append_copy(u8"warning: flag: '--vim-file-bufnr="sv);
+      string8_view number{to_string8(std::to_string(bufnr))};
+      out.append_copy(number);
```

Comment:
Must fix: I think this creates a dangling pointer. The string returned by `to_string8` is deallocated after line 183, but we use a pointer to the deallocated data on line 184. See ASAN report below.

Easy fix: call `out.append_decimal_integer` instead:
```suggestion
      out.append_decimal_integer(bufnr);
```

ASAN report:
```
==96002==ERROR: AddressSanitizer: stack-use-after-scope on address 0x7fff9748bd40 at pc 0x7f3e80038f50 bp 0x7fff9748b590 sp 0x7fff9748ad38
READ of size 1 at 0x7fff9748bd40 thread T0
    #0 0x7f3e80038f4f in __interceptor_memmove ../../../../src/libsanitizer/sanitizer_common/sanitizer_common_interceptors.inc:773
    #1 0x5595c75da0f0 in char8_t* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<char8_t>(char8_t const*, char8_t const*, char8_t*) /usr/include/c++/9/bits/stl_algobase.h:386
    #2 0x5595c75d6e2b in char8_t* std::__copy_move_a<false, char8_t const*, char8_t*>(char8_t const*, char8_t const*, char8_t*) /usr/include/c++/9/bits/stl_algobase.h:404
    #3 0x5595c75d0d8b in char8_t* std::__copy_move_a2<false, char8_t const*, char8_t*>(char8_t const*, char8_t const*, char8_t*) /usr/include/c++/9/bits/stl_algobase.h:440
    #4 0x5595c75c7868 in char8_t* std::copy<char8_t const*, char8_t*>(char8_t const*, char8_t const*, char8_t*) /usr/include/c++/9/bits/stl_algobase.h:474
    #5 0x5595c94b9b22 in quick_lint_js::output_stream::append_copy(std::basic_string_view<char8_t, std::char_traits<char8_t> >) /home/strager/Projects/quicklint-js/src/output-stream.cpp:34
    #6 0x5595c94b5c1a in quick_lint_js::options::dump_errors(quick_lint_js::output_stream&) const /home/strager/Projects/quicklint-js/src/options.cpp:184
    #7 0x5595c8144ccb in TestBody /home/strager/Projects/quicklint-js/test/test-options.cpp:452
    #8 0x5595c916a740 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:2607
    #9 0x5595c9144bf5 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:2662
    #10 0x5595c90d68bb in testing::Test::Run() /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:2682
    #11 0x5595c90d87eb in testing::TestInfo::Run() /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:2861
    #12 0x5595c90db9f1 in testing::TestSuite::Run() /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:3015
    #13 0x5595c910dbc4 in testing::internal::UnitTestImpl::RunAllTests() /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:5855
    #14 0x5595c9171faa in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:2607
    #15 0x5595c914a24d in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:2662
    #16 0x5595c9105508 in testing::UnitTest::Run() /home/strager/Projects/quicklint-js/vendor/googletest/googletest/src/gtest.cc:5438
    #17 0x5595c90b0676 in RUN_ALL_TESTS() /home/strager/Projects/quicklint-js/vendor/googletest/googletest/include/gtest/gtest.h:2490
    #18 0x5595c90b074a in main /home/strager/Projects/quicklint-js/vendor/googletest/googlemock/src/gmock_main.cc:70
    #19 0x7f3e7f0ec0b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x240b2)
    #20 0x5595c72cff6d in _start (/home/strager/Projects/quicklint-js/build-debug/test/quick-lint-js-test+0x6a31f6d)

Address 0x7fff9748bd40 is located in stack of thread T0 at offset 1040 in frame
    #0 0x5595c94b4c03 in quick_lint_js::options::dump_errors(quick_lint_js::output_stream&) const /home/strager/Projects/quicklint-js/src/options.cpp:148

  This frame has 15 object(s):
    [32, 48) 'number' (line 183)
    [64, 80) '<unknown>'
    [96, 112) '<unknown>'
    [128, 168) '__for_begin' (line 181)
    [208, 248) '__for_end' (line 181)
    [288, 328) '__for_begin' (line 190)
    [368, 408) '__for_end' (line 190)
    [448, 488) '__for_begin' (line 197)
    [528, 568) '__for_end' (line 197)
    [608, 648) '__for_begin' (line 203)
    [688, 728) '__for_end' (line 203)
    [768, 824) '<unknown>'
    [864, 920) '<unknown>'
    [960, 992) '<unknown>'
    [1024, 1056) '<unknown>' <== Memory access at offset 1040 is inside this variable
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-use-after-scope ../../../../src/libsanitizer/sanitizer_common/sanitizer_common_interceptors.inc:773 in __interceptor_memmove
Shadow bytes around the buggy address:
  0x100072e89750: f2 f2 f2 f2 00 00 00 00 00 f2 f2 f2 f2 f2 00 00
  0x100072e89760: 00 00 00 f2 f2 f2 f2 f2 00 00 00 00 00 f2 f2 f2
  0x100072e89770: f2 f2 00 00 00 00 00 f2 f2 f2 f2 f2 00 00 00 00
  0x100072e89780: 00 f2 f2 f2 f2 f2 00 00 00 00 00 00 00 f2 f2 f2
  0x100072e89790: f2 f2 00 00 00 00 00 00 00 f2 f2 f2 f2 f2 f8 f8
=>0x100072e897a0: f8 f8 f2 f2 f2 f2 f8 f8[f8]f8 f3 f3 f3 f3 00 00
  0x100072e897b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100072e897c0: 00 00 00 00 00 00 00 00 f1 f1 f1 f1 01 f2 01 f2
  0x100072e897d0: 01 f2 01 f2 00 00 00 f2 00 00 00 f2 00 00 00 f2
  0x100072e897e0: 00 00 00 f2 00 00 00 f2 00 00 00 f2 00 00 00 f2
  0x100072e897f0: 00 00 00 f2 00 00 00 f2 00 00 00 f2 00 00 00 f2
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==96002==ABORTING
```